.github/workflows/ci.yml

```
@@ -106,71 +99,60 @@ jobs:
+          FAIL=0
+          [ "${{ needs.test.result }}"  != "success" ] && echo "❌ Tests failed" && FAIL=1
+          [ "${{ needs.e2e.result }}"   != "success" ] && echo "❌ E2E failed"   && FAIL=1
+          [ "${{ needs.build.result }}" != "success" ] && echo "❌ Build failed" && FAIL=1
+          [ "$FAIL" -eq 0 ] && echo "✅ All checks passed"
+          exit $FAIL
+
+      - name: PR status comment
         if: github.event_name == 'pull_request'
         uses: actions/github-script@v7
         with:
           script: |
```

actionlint (1.7.12)

[error] "github.head_ref" is potentially untrusted. avoid using it directly in inline scripts. instead, pass it through an environment variable. see https://docs.github.com/en/actions/reference/security/secure-use#good-practices-for-mitigating-script-injection-attacks for more details

(expression)